code/script.js

```diff
@@ -222,104 +227,189 @@ const generateBoard = () => {
+  const value = questions.value // This variable stores the actual value of the quesiton we've selected in the dropdown.
+    
+    if (category === 'hair') {
+      console.log('hair:', value)
+      currentQuestion = { 
+        category: category,
+        value: value
+      }
+    }
+  
+    else if (category === 'eyes') {
+      console.log('eyes:', value)
+      currentQuestion = { 
+        category: category,
+        value: value
+      }
+    }
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+    else if (category === 'accessories') {
+      console.log('accessories:', value)
+      currentQuestion = { 
+        category: category,
+        value: value 
+      }
+    }
 
-  currentQuestion = {
-    category: category,
-    // value: value
+    else if (category === 'other') {
+      console.log('other:', value)
+      currentQuestion = {
+        category: category,
+        value: value 
+      }
+     }
   }
-}
 
-// This function should be invoked when you click on 'Find Out' button.
+// Function: checkQuestion is invoked when pressing the findOutButton, and leads to the filter function. 
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  console.log('Please check if the secret person has these attributes!')
+  const { category, value } = currentQuestion 
+  
+  let keep
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+    console.log(category, value) // Daniels förslag 
+    if (value === secret.hair || value === secret.eyes) {
+      keep = true
+      filterCharacters(true) 
+      console.log('You are on the right track!')
+    }
+    else {
+      keep = false
+      filterCharacters() 
+      console.log('You are not on the right track!')
+    }
+  } 
+  
+  else if (category === 'accessories' || category === 'other') {
+    console.log(category, value) // Daniels förslag 
+    if (secret.accessories.includes(value) || secret.other.includes(value)) { //this says includes instead of === since you either have the accessories/other or you don't. 
+      keep = true
+      filterCharacters(true) //vad ska man ha inom dessa paranteser? 
+      console.log('You are not on the right track')
+    }
+    else {
+      keep = false
+      filterCharacters() //samma som ovan 
+      console.log('You are not on the right track!')
+    }
   }
 }
 
-// It'll filter the characters array and redraw the game board.
+// Function: filterCharachters filters the charachters in the array and redraws the board. The redraw will depend on the user's question.
```

Comment:
Similarly to `selectQuestion()`, `filterCharacters()` could be shortened a bit by having the conditionals for eyes and hair together by using the `category` variable as well, e.g. : 
if (category === 'accessories') {
    if (keep) {
      //keep as you have written the code 
    } 
    else {
       //keep as you have written the code
    }
  } 
  else if (category === 'other') {
    if (keep) {
       //keep as you have written the code
    } 
    else {
       //keep as you have written the code
    }
  } 
  else {
    if (keep) {
      alert(`Yes, the person has ${value} ${category}! Keep all people with ${value} ${category}`);
      charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
    } 
    else {
      alert(`No, the person doesn't have ${value} ${category}! Remove all people with ${value} ${category}`);
      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
    }